compiler/luci/pass/src/RemoveRedundantTranspose.cpp

```diff
@@ -0,0 +1,125 @@
+bool check_perm(const luci::CircleConst *first_perm, const luci::CircleConst *second_perm)
+{
+  assert(first_perm->rank() == 1);
+  assert(second_perm->rank() == 1);
+  assert(second_perm->size<loco::DataType::S32>() == first_perm->size<loco::DataType::S32>());
+  for (int32_t i = 0; i < static_cast<int32_t>(first_perm->size<loco::DataType::S32>()); i++)
+  {
+    if (first_perm->at<loco::DataType::S32>(second_perm->at<loco::DataType::S32>(i)) != i)
+      return false;
+  }
+  return true;
+}
+
+bool remove_consecutive_transpose_function(luci::CircleNode *node)
+{
+  if (node->opcode() != luci::CircleOpcode::TRANSPOSE)
+    return false;
+
+  auto pred_node = static_cast<luci::CircleNode *>(node->arg(0));
+  if (pred_node->opcode() != luci::CircleOpcode::TRANSPOSE)
+    return false;
+  if (loco::succs(pred_node).size() != 1)
+    return false;
+
+  auto pred_perm = dynamic_cast<luci::CircleConst *>(node->arg(1));
+  if (pred_perm == nullptr)
+    return false;
+
+  auto main_perm = dynamic_cast<luci::CircleConst *>(pred_node->arg(1));
+  if (main_perm == nullptr)
+    return false;
+
+  auto main_node = static_cast<luci::CircleNode *>(pred_node->arg(0));
+  if (check_perm(pred_perm, main_perm))
+  {
+    replace(node).with(main_node);
+    pred_node->drop();
+  }
+  else
+  {
+    auto g = main_perm->graph();
+    auto new_const_node = g->nodes()->create<luci::CircleConst>();
+
+    new_const_node->dtype(loco::DataType::S32);
+    new_const_node->rank(main_perm->rank());
+    uint32_t dim_size = 1;
+    for (uint32_t i = 0; i < new_const_node->rank(); ++i)
+    {
+      new_const_node->dim(i) = main_perm->dim(i);
+      dim_size *= main_perm->dim(i).value();
+    }
+    new_const_node->size<loco::DataType::S32>(dim_size);
+    new_const_node->shape_status(luci::ShapeStatus::VALID);
+    for (uint32_t i = 0; i < main_perm->size<loco::DataType::S32>(); i++)
+    {
+      new_const_node->at<loco::DataType::S32>(i) =
+          pred_perm->at<loco::DataType::S32>(main_perm->at<loco::DataType::S32>(i));
+    }
+    replace(main_perm).with(new_const_node);
+    replace(node).with(pred_node);
+  }
+  node->drop();
+  return true;
+}
+
+/**
+ *  BEFORE
+ *         |
+ *   [CircleInput]    [CircleConst]
+ *           \              /
+ *           [CircleTranspose]  [CircleConst]
+ *                   \               /
+ *                   [CircleTranspose]
+ *                           |
+ *
+ *  AFTER
+ *      <Optional Case>
+ *
+ *          |                 |               |
+ *    [CircleInput]     [CircleConst]         |
+ *           \               /           or   |  [Remove all]
+ *           [CircleTranspose]                |
+ *                   |                        |
+ *
+ */
+bool RemoveRedundantTransposePass::run(loco::Graph *g)
+{
+  bool changed = false;
+  for (auto node : loco::active_nodes(loco::output_nodes(g)))
+  {
+    auto circle_node = static_cast<luci::CircleNode *>(node);
```

Comment:
It would be more safe.
```suggestion
    auto circle_node = dynamic_cast<luci::CircleNode *>(node);
```

Reply:
If `circle_node` is `nullptr`, segmentation fault occurs.
When casting `loco::Node` to `luci::CircleNode`, `loco::must_cast` is used normally.
```suggestion
    auto circle_node = loco::must_cast<luci::CircleNode *>(node);
```

Reply:
I think @llFreetimell is right.

Reply:
I update with using `loco::must_cast` on this.